### PR TITLE
Require Node version at least 16.16 LTS for development

### PR DESCRIPTION
**Pull request by andre4i**
## Description

ADO:3307

This will match the recommended major version from `.nvmrc` and will require all developers to upgrade. Instructions at https://github.com/microsoft/FluidFramework/pull/14304

### CODE REVIEW

File: azure/packages/azure-local-service/package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
 		"typescript": "~4.5.5"
 	},
 	"engines": {
-		"node": ">=14.0.0"
```
Comment (CraigMacomber) [inline azure/packages/azure-local-service/package.json:55]: This package is one we actually publish, so changing this will break users of it, not just suggest that developers upgrade. I'd recommend not changing this for now, and when we do change it, do it on next and announce it to consumers in advance.

File: build-tools/packages/build-cli/package.json [hunk 2]
```diff
@@ -143,7 +143,7 @@
 		"typescript": "~4.5.5"
 	},
 	"engines": {
-		"node": ">=14.0.0"
```
Comment (CraigMacomber) [inline build-tools/packages/build-cli/package.json:146]: This is also a breaking change. Thats probably fine to require the same version of node for running our build tools as we do for development.

File: server/tinylicious/package.json [hunk 3]
```diff
@@ -106,7 +106,7 @@
 		"typescript": "~4.5.5"
 	},
 	"engines": {
-		"node": ">=14.0.0"
```
Comment (CraigMacomber) [inline server/tinylicious/package.json:109]: This is a breaking change to a published package. If we do this it probably shouldn't be part of this PR.
Also we probably don't need to go all the way to 16.16 for this either.

File: azure/packages/azure-local-service/package.json (hunk 1 above)
Comment (andre4i) [reply azure/packages/azure-local-service/package.json:55]: I will revert this (and other package.json files) but keep the root package.json in this change, that one shouldn't affect users of the package.

File: build-tools/packages/build-cli/package.json (hunk 2 above)
Comment (CraigMacomber) [reply build-tools/packages/build-cli/package.json:146]: Same applies to the other changes in buil-tools

**Comment by CraigMacomber** [comment]
Description implies this is just intended to impact developers of FF, but due to changing the supported node version for released packages (not just unpublished release group roots which we use to suggest developers use a specific version), this actually impacts users of our packages and thus has several breaking changes.

I'd recommend bumping the single engine versions in the release group root package.json files as a way for force our devs to upgrade. If that goes well, update the build tools to match, then coordinate with consumers regarding the breaking change to the server package's node version requirements (which likely need to go into the next branch).

**Review by CraigMacomber — APPROVED** [review]
